README.md

```diff
@@ -0,0 +1,64 @@
+# Diplomat
+
+Diplomat is a conversational agent framework for goal-oriented group discussion. Recent work in human-computer interaction has explored the use of conversational agents as facilitators for group goal-oriented discussions. Inspired by this work and by the apparent lack of tooling available to support it, we created Diplomat, a Python-based framework for building conversational agent facilitators. Diplomat is designed to support simple specification of agent functionality as well as customizable integration with online chat services. We document a preliminary user study we conducted to help inform the design of Diplomat.
+
+## How to run Diplomat? 
+Diplomat has currently been configured to run seamlessly with Slack. 
+
+#### TLDR:
+Setup your python environment by running 
+`pip3 install requirements.txt`.
+
+You need the following to run an instance of diplomat: 
+1. Configuration file containing information of which features to run. `(conf.json)`
+2. A Slackbot token generated by Slack.
+3. Name of the channel you want Diplomat to monitor group discussions in.
+4. User ID of your bot. 
+
+Once you have these, simply run diplomat by using the following command: 
+`python3 slack_integration -p conf.json -t <slackbot-token> -c <channel-id> -b <bot-id>` 
+
+### Detailed steps for a developing your custom plugin:
+#### Diplomat's architecture
+Diplomat's software consists of the following classes: 
+
+* `Class ChatServiceToBotIntegrator`: This is the main class which runs the diplomat code. 
+It loads classes (plugins) from the `feature_plugins.py` file into memory along with the user defined `conf.json`. 
+It then loads the required plugin and passes the respective configuration to the plugin. This class has 2 main functions 
+which can be overridden. All chatbots inherit from this class. 
+    1. `request_transcript_and_convert_to_message_list`: Receive transcript from your chat service (e.g. Slack, Teams)
+    and convert it into a list of messages.
+    2. `post_chatbot_interventions`: Send messages to your chat service.     
+    <br>
+    * Sample Subclass: `Class SlackToBotIntegrator`: This is a sample slackbot integration class which inherits from the ChatService class 
+defined above. The class contains APIs to receive a transcript from slack and APIs to post interventions to slack. 
+
+* `Class FeaturePlugin`: This is the super class which will be inherited by all plugins. The class extends 1 API called
+`generate_interventions`. The API takes the `chat_transcript` and `author_id` as input. The class also receives the 
+configuration parameters passed into the constructor. 
+
+    * Sample Plugin: `class OverSpeakingPlugin`:  You can find this class in `feature_plugins.py`. We see that config is passed 
+as a parameter to the `__init__` class. The plugin has a static method called `get_overspeaking_authors`. Ultimately, the plugin
+implements the `generate_interventions class.` 
+ 
+
+#### Creating the config file: 
```

Comment:
Might also want to mention that the presence or absence of a feature in the configuration file determines whether that feature will be enabled for a particular agent.